discovery/file/file.go

```diff
@@ -256,6 +256,11 @@ func readFile(filename string) ([]*config.TargetGroup, error) {
 	}
 
 	for i, tg := range targetGroups {
+		if tg == nil {
+			log.Errorf("nil target group item found from: %s", filename)
+			continue
```

Comment:
I don't think we should accept a broken file. This also doesn't increment `fileSDReadErrorsCount`. IMHO we should just return an error here, the caller will take care of updating the metric and logging the error using the correct logger (instead of using the package logger).